lib/open-uri.rb

```diff
@@ -237,6 +237,11 @@ def OpenURI.open_loop(uri, options) # :nodoc:
           options = options.dup
           options.delete :http_basic_authentication
         end
+        if !(uri.scheme == redirect.scheme && uri.host == redirect.host && uri.port == redirect.port)
```

Comment:
I apologize for my misunderstanding of the Same-Origin Policy.
The examples I mentioned do violate the Same-Origin Policy.
- ref: https://developer.mozilla.org/en-US/docs/Web/Security/Same-origin_policy

However, it might be worth discussing whether we need to support these scenarios, considering backward compatibility and specific use cases.